mbed_host_tests/host_tests_runner/mbed_base.py

```diff
@@ -92,10 +113,69 @@ def copy_image(self, image_path=None, disk=None, copy_method=None, port=None):
             copy_method = self.copy_method
         if not port:
             port = self.port
+        if not retry_copy:
+            retry_copy = self.retry_copy
+
+        flash_done = False
+        target_id = self.target_id
+
+        for count in range(0, retry_copy):
+            if flash_done:
+                break
+            initial_remount_count = get_remount_count(disk)
+
+            # Call proper copy method
+            result = self.copy_image_raw(image_path, disk, copy_method, port)
+            sleep(self.program_cycle_s)
+            if not result:
+                continue
+
+            if not target_id:
+                self.logger.prn_wrn("Target ID not found: Skipping flash check and retry")
+                break
+
+            bad_files = Set(['FAIL.TXT'])
+
+                # Re-try at max 5 times with 0.5 sec in delay
+            for i in range(5):
+                # mbed_lstools.create() should be done inside the loop. Otherwise it will loop on same data.
+                mbeds = mbed_lstools.create()
+                mbeds_by_tid = mbeds.list_mbeds_by_targetid()   # key: target_id, value mbedls_dict()
+
+                if target_id in mbeds_by_tid:
+                    if 'mount_point' in mbeds_by_tid[target_id] and mbeds_by_tid[target_id]['mount_point']:
```

Comment:
Need to check. But here ```if mbeds_by_tid[target_id]``` check means if directory is not empty. If we pass ```if 'mount_point' in mbeds_by_tid[target_id]``` then ```if mbeds_by_tid[target_id]``` is always true.